code/modules/reagents/Chemistry-Reagents.dm

```diff
@@ -9677,26 +9677,47 @@ var/global/list/tonio_doesnt_remove=list("tonio", "blood")
 /datum/reagent/ectoplasm
 	name = "Ectoplasm"
 	id = ECTOPLASM
-	description = "Pure, distilled spooky"
+	description = "Pure, distilled spooky."
 	reagent_state = REAGENT_STATE_LIQUID
 	color = "#21d389b4"
 	density = 0.05
-
+	custom_metabolism = 0.01
+	var/spookvision = FALSE
+	
 /datum/reagent/ectoplasm/on_mob_life(var/mob/living/M)
 	if(..())
 		return 1
+
+	if(tick >= 30 && volume >= 1 && !spookvision) //ghostsight after 1m and having more than 1u inside
+		spookvision = TRUE
+		to_chat(M, "<span class='notice'>You start seeing through the veil!</span>")
+		M.see_invisible = SEE_INVISIBLE_OBSERVER
+		M.see_invisible_override = SEE_INVISIBLE_OBSERVER
+
+	if(spookvision && volume < 1)
+		spookvision = FALSE
+		to_chat(M, "<span class='notice'>Your otherworldly sight suddenly vanishes!</span>")
+		M.see_invisible = initial(M.see_invisible)
+		M.see_invisible_override = 0
+
 	if(isskellington(M) || isskelevox(M) || islich(M))	//Slightly better than DD for spooks
 		playsound(M, 'sound/effects/rattling_bones.ogg', 100, 1)
 		if(M.getOxyLoss())
 			M.adjustOxyLoss(-3)
+			holder.remove_reagent(ECTOPLASM, 0.1)
```

Comment:
this is actually a substantial buff to ectoplasm as a healing chem for spooky races because it lasts twice as long and it's a pretty robust healer

on the other hand it's hard to mass produce so maybe it's fine - I won't hold it up over this, just a word of caution

Reply:
That's why the dedicated 0.1u removal on spookraces, before it processed 0.2 per tick but healed all damages simultaneously, it now processes at 0.01 per tick but removes 0.1u per damage type, meaning that healing more than 1 damage type is more expensive metabolismwise than before.